soh/soh/Enhancements/mods/modhooks.cpp

```diff
@@ -63,6 +64,33 @@ void RegisterAutoSaveOnReceiveItem() {
     });
 }
 
+void RegisterRupeeDash() {
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnPlayerUpdate>([]() {
+        if (!CVarGetInteger("gRupeeDash", 0)) {
+            return;
+        }
+        
+        // Initialize Timer
+        static uint16_t rupeeDashTimer = 0;
+        uint16_t rdmTime = CVarGetInteger("gDashInterval", 5) * 20;
+        
+        // Did time change by DashInterval?
+        if (rupeeDashTimer >= rdmTime) {
+            rupeeDashTimer = 0;
+            if (gSaveContext.rupees > 0) {
+                Rupees_ChangeBy(-1);
+            } else {
+                if (gSaveContext.health > 0) {
+                    gSaveContext.health = gSaveContext.health - 16;
```

Comment:
@dcvz im not taking suggestions at this time... lmao jk.

Solid point, i can double check but im pretty sure i had a quarter heart less than full at i just died rather than crashing ot anything